src/core/datastore/deltaset.js

```diff
@@ -0,0 +1,159 @@
+import isArray from 'lodash/isArray';
+import { format } from 'url';
+import { Query } from '../query';
+import { KinveyRequest, RequestMethod, AuthType } from '../request';
+import { Client } from '../client';
+import { KinveyError } from '../errors';
+import { repositoryProvider } from './repositories';
+import { buildCollectionUrl } from './repositories/utils';
+import {
+  stripTagFromCollectionName,
+  generateEntityId,
+  queryCacheCollectionName,
+  xKivneyRequestStartHeader
+} from './utils';
+
+function getDeltaSetQuery(collectionNameWithTag, query) {
+  const serializedQuery = query ? query.toString() : '';
+
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      const queryCacheQuery = new Query()
+        .equalTo('collectionName', collectionNameWithTag)
+        .and()
+        .equalTo('query', serializedQuery);
+      return offlineRepo.read(queryCacheCollectionName, queryCacheQuery)
+        .then((deltaSetQueryDocs = []) => {
+          if (deltaSetQueryDocs.length > 0) {
+            return deltaSetQueryDocs[0];
+          }
+
+          return {
+            _id: generateEntityId(),
+            collectionName: collectionNameWithTag,
+            query: serializedQuery
+          };
+        });
+    });
+}
+
+function updateDeltaSetQuery(deltaSetQuery, response) {
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      deltaSetQuery.lastRequest = response.headers.get(xKivneyRequestStartHeader);
+      return offlineRepo.update(queryCacheCollectionName, deltaSetQuery);
+    });
+}
+
+function readDocs(collectionNameWithTag, query) {
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      return offlineRepo.read(collectionNameWithTag, query);
+    });
+}
+
+function deleteDocs(collectionNameWithTag, deleted = []) {
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      if (isArray(deleted) && deleted.length > 0) {
+        const deletedIds = deleted.map((doc) => doc._id);
+        const deleteQuery = new Query().contains('_id', deletedIds);
+        return offlineRepo.delete(collectionNameWithTag, deleteQuery)
+      }
+
+      return deleted;
+    });
+}
+
+function updateDocs(collectionNameWithTag, changed = []) {
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      if (isArray(changed) && changed.length > 0) {
+        return offlineRepo.update(collectionNameWithTag, changed)
+      }
+
+      return changed;
+    });
+}
+
+function makeDeltaSetRequest(collectionNameWithTag, deltaSetQuery, query, options) {
+  const collectionName = stripTagFromCollectionName(collectionNameWithTag);
+  const client = Client.sharedInstance();
+  const request = new KinveyRequest({
```

Comment:
@georgiwe I don't completely understand your comment. What `KinveyRequest.execute()` call are you referring to? Is there an example.

I agree in principle with your point, though. It seems like a bunch of these options (`skipBL`, `timeout` etc) would apply to all data requests, so some way to avoid listing them out one-by-one will be useful.

Reply:
I mean like [here](https://github.com/Kinvey/js-sdk/blob/master/src/core/live/collection/live-collection-manager.js#L70). It basically takes care of all these defaults which are used over 90% of the time, extracting the data, etc. Just getting rid of boilerplate.